### ISSUE TITLE
TypeError: i.substring is not a function

### ISSUE THREAD

**Issue by matt77vinyl** [adobe-dmeservices/custom-metadata#23]
Experiencing a reproducible error, occurs when I click on what so far seems to be one particular pdf in the content tab, leads immediately to red alert "something went wrong" appearing in the custom metadata panel --- the copy function from the error shows "TypeError: i.substring is not a function".  I thereafter cannot utilize the custom metadata panel (it's completely empty) for any pdfs, until I reopen Bridge (using v12.0.1 btw on OS Big Sur), but then recurs each time I select the particular pdf.  I deleted the offending pdf and made a new version of it, but after some time where it does allow me to enter metadata like usual, the error recurs.
I tried purging all cache which did nothing.

thanks in advance for any suggestions.
Matt

**Comment by jameslockman** [comment]
Can you please share a file that exhibits this behavior?
Can you share what version of the panel you’re using?

--
James Lockman
Senior Manager, Digital Media Services
Adobe
***@***.******@***.***>
(917) 284-2268 Mobile
Sent from his iPhone

On Apr 19, 2022, at 11:28 PM, matt77vinyl ***@***.***> wrote:

﻿

Experiencing a reproducible error, occurs when I click on what so far seems to be one particular pdf in the content tab, leads immediately to red alert "something went wrong" appearing in the custom metadata panel --- the copy function from the error shows "TypeError: i.substring is not a function". I thereafter cannot utilize the custom metadata panel (it's completely empty) for any pdfs, until I reopen Bridge (using v12.0.1 btw on OS Big Sur), but then recurs each time I select the particular pdf. I deleted the offending pdf and made a new version of it, but after some time where it does allow me to enter metadata like usual, the error recurs.
I tried purging all cache which did nothing.

thanks in advance for any suggestions.
Matt

**Comment by matt77vinyl** [comment]
thank you -- panel version 1.5.0
file attached (I had to compress/resize for upload.
[mwd0000000002smaller.pdf](https://github.com/adobe-dmeservices/custom-metadata/files/8521733/mwd0000000002smaller.pdf)

**Comment by bhunut-adobe** [comment]
@matt77vinyl  Can you provide us with the exported JSON of the metadata view? I'm unable to replicate this issue so far and wondering if it specific to reading a particular field that is throwing this error.

**Comment by matt77vinyl** [comment]
thank you --- I haven't traced it to one field yet, but it does seem to allow me to do some metadata editing for awhile before the error recurs.  I'll do a few more trials to see if I can pinpoint a field.  
Attaching JSON
[share.json.zip](https://github.com/adobe-dmeservices/custom-metadata/files/8526659/share.json.zip)
.

**Comment by matt77vinyl** [comment]
after more testing, I did find that it's when I enter data into my field for "event date" that the error occurs.  Is there something I can change to allow that field to work?

**Comment by bhunut-adobe** [comment]
@matt77vinyl  Sorry for the late response. I look deeper into this. At the moment, our date field doesn't support "Array of Date."  We will add this as a priority to do for the next major release.

<img width="1031" alt="image" src="https://user-images.githubusercontent.com/26125757/181615197-2344133d-8752-469e-93f3-3f204753fa61.png">

```
<dc:date>
  <rdf:Seq>
     <rdf:li>1978-06-02T17:00:00.000Z</rdf:li>
  </rdf:Seq>
</dc:date>
```

As a workaround, I recommend you change DC:Date to the MultiText field instead. You won't have Date Parsing, but it should allow you to read and modify this field.

<img width="397" alt="image" src="https://user-images.githubusercontent.com/26125757/181615135-e2506805-df61-414b-984b-eabed8b4b51f.png">

See attached for modified JSON of the change.
[share 3.json.zip](https://github.com/adobe-dmeservices/custom-metadata/files/9213407/share.3.json.zip)

**Comment by jameslockman** [comment]
Closed by 1.7.0 release, which adds Multi Date field.
